ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/electra/block/BlockProcessorElectraTest.java

```diff
@@ -0,0 +1,144 @@
+  }
+
+  @Test
+  public void verifiesOutstandingEth1DepositsAreProcessed() {
+    final BeaconState state =
+        createBeaconState()
+            .updated(
+                mutableState -> {
+                  final UInt64 eth1DepositCount = UInt64.valueOf(25);
+                  mutableState.setEth1Data(
+                      new Eth1Data(
+                          dataStructureUtil.randomBytes32(),
+                          eth1DepositCount,
+                          dataStructureUtil.randomBytes32()));
+                  final UInt64 eth1DepositIndex = UInt64.valueOf(13);
+                  mutableState.setEth1DepositIndex(eth1DepositIndex);
+                  final UInt64 depositReceiptsStartIndex = UInt64.valueOf(20);
+                  MutableBeaconStateElectra.required(mutableState)
+                      .setDepositReceiptsStartIndex(depositReceiptsStartIndex);
+                });
+
+    final BeaconBlockBody body =
+        dataStructureUtil.randomBeaconBlockBody(
+            // 20 - 13 = 7
+            builder -> builder.deposits(dataStructureUtil.randomSszDeposits(7)));
+
+    getBlockProcessor(state).verifyOutstandingDepositsAreProcessed(state, body);
+  }
+
+  @Test
+  public void
+      verifiesNoOutstandingEth1DepositsAreProcessedWhenFormerDepositMechanismHasBeenDisabled() {
+    final BeaconState state =
+        createBeaconState()
+            .updated(
+                mutableState -> {
+                  final UInt64 eth1DepositCount = UInt64.valueOf(25);
+                  mutableState.setEth1Data(
+                      new Eth1Data(
+                          dataStructureUtil.randomBytes32(),
+                          eth1DepositCount,
+                          dataStructureUtil.randomBytes32()));
+                  final UInt64 eth1DepositIndex = UInt64.valueOf(20);
+                  mutableState.setEth1DepositIndex(eth1DepositIndex);
+                  final UInt64 depositReceiptsStartIndex = UInt64.valueOf(20);
+                  MutableBeaconStateElectra.required(mutableState)
+                      .setDepositReceiptsStartIndex(depositReceiptsStartIndex);
+                });
+
+    final BeaconBlockBody body =
+        dataStructureUtil.randomBeaconBlockBody(
+            // 20 - 20 = 0
+            modifier -> modifier.deposits(dataStructureUtil.randomSszDeposits(0)));
+
+    getBlockProcessor(state).verifyOutstandingDepositsAreProcessed(state, body);
+  }
+
+  @Test
+  public void processesDepositReceipts() {
+    final BeaconStateElectra preState =
+        BeaconStateElectra.required(
+            createBeaconState()
+                .updated(
+                    mutableState ->
+                        MutableBeaconStateElectra.required(mutableState)
+                            .setDepositReceiptsStartIndex(
+                                SpecConfigElectra.UNSET_DEPOSIT_RECEIPTS_START_INDEX)));
+    final int numberOfValidators = preState.getValidators().size();
```

Comment:
maybe something like `preElectraValidators` to highlight it's pre-transition number?